### PR TITLE
fix borrows

### PR DESCRIPTION
**Pull request by Emilgardis**
- specify lifetime always to ensure bound is always correct
- add `deser_borrow` feature flag

### CODE REVIEW

**Comment by Emilgardis** [comment]
It's unclear to me if this should be in default or not.

Feature flags are supposed to be additive, and the addition of `#[serde(borrow)]` could limit usage. As such, this should probably be in default features, but that means that dependencies need to make sure that they specify `default-features = false`

File: src/helix/endpoints/channels/get_vips.rs [hunk 1]
```diff
@@ -46,18 +46,20 @@ use helix::RequestGet;
...
-    #[serde(borrow)]
+    #[cfg_attr(feature = "deser_borrow", serde(borrow = "'a"))]
     pub broadcaster_id: Cow<'a, types::UserIdRef>,
     /// Filters the list for specific VIPs. To specify more than one user, include the user_id parameter for each user to get. For example, &user_id=1234&user_id=5678. The maximum number of IDs that you may specify is 100. Ignores those users in the list that aren’t VIPs.
     #[cfg_attr(feature = "typed-builder", builder(default))]
-    #[serde(borrow)]
+    #[cfg_attr(feature = "deser_borrow", serde(borrow = "'a"))]
+    // FIXME: This is essentially the same as borrow, but worse
+    #[cfg_attr(not(feature = "deser_borrow"), serde(bound(deserialize = "'de: 'a")))]
     pub user_id: Cow<'a, [&'a types::UserIdRef]>,
```
Comment (Emilgardis) [inline src/helix/endpoints/channels/get_vips.rs:56]: this, and the others means that they can't be deserialized with `DeserializeOwned`

File: Cargo.toml [hunk 2]
```diff
@@ -83,6 +83,8 @@ helix = [
     "dep:hyper",
 ]
 
+deser_borrow = []
```
Comment (Nerixyz) [inline Cargo.toml:86]: This feature should be documented in `lib.rs`.

**Review by Nerixyz — CHANGES_REQUESTED** [review]
> It's unclear to me if this should be in default or not.

I'm really unsure. I think having it in the default features is fine, but it should be mentioned in the changelog.

File: src/helix/endpoints/channels/get_vips.rs (hunk 1 above)
Comment (Emilgardis) [reply src/helix/endpoints/channels/get_vips.rs:56]: i.e

```
#[derive(serde::Deserialize)]
struct Foo {
    bar: GetVipsRequest<'static>,
}
```

is not usable with `DeserializeOwned`

**Review by Nerixyz — APPROVED** [review]

**Comment by Emilgardis** [comment]
bors r=Nerixyz

**Pull request merged**